frontend/src/features/SideWindow/MissionForm/ActionForm/shared/FleetSegmentsField.tsx

```diff
@@ -0,0 +1,127 @@
+import { Field, Label, noop, Select, SingleTag, TagGroup } from '@mtes-mct/monitor-ui'
+import { useField } from 'formik'
+import { useCallback, useEffect, useMemo } from 'react'
+
+import { useGetFleetSegmentsQuery } from '../../../../../api/fleetSegment'
+import { getFaoZonesFromSpeciesOnboard } from '../../../../../domain/entities/vessel/riskFactor'
+import { getVesselRiskFactor } from '../../../../../domain/use_cases/vessel/getVesselRiskFactor'
+import { useMainAppDispatch } from '../../../../../hooks/useMainAppDispatch'
+// import { useMainAppSelector } from '../../../../../hooks/useMainAppSelector'
+import { useNewWindow } from '../../../../../ui/NewWindow'
+import { FieldsetGroup } from '../../FieldsetGroup'
+
+import type { MissionAction } from '../../../../../domain/types/missionAction'
+import type { MissionActionFormValues } from '../../types'
+import type { Option } from '@mtes-mct/monitor-ui'
+
+export function FleetSegmentsField() {
+  const [input, , helper] = useField<MissionActionFormValues['segments']>('segments')
+
+  const { newWindowContainerRef } = useNewWindow()
+
+  const dispatch = useMainAppDispatch()
+
+  const getFleetSegmentsApiQuery = useGetFleetSegmentsQuery()
+
+  const fleetSegmentsAsOptions: Option<MissionAction.FleetSegment>[] = useMemo(() => {
+    if (!getFleetSegmentsApiQuery.data) {
+      return []
+    }
+
+    return getFleetSegmentsApiQuery.data.map(({ faoAreas, segment, segmentName }) => ({
+      label: `${segment} - ${segmentName}`,
+      value: {
+        faoAreas: faoAreas || [],
+        segment,
+        segmentName: segmentName || undefined
+      }
+    }))
+  }, [getFleetSegmentsApiQuery.data])
+
+  const add = useCallback(
+    (newSegment: MissionAction.FleetSegment | undefined) => {
+      if (!newSegment) {
+        return
+      }
+
+      const nextSegments: MissionAction.FleetSegment[] = [...(input.value || []), newSegment]
+
+      helper.setValue(nextSegments)
+    },
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [input.value]
+  )
+
+  useEffect(() => {
+    const getRiskFactor = async () => {
+      try {
+        const riskFactor = await dispatch(getVesselRiskFactor(''))
```

Comment:
The param is missing ?

Reply:
It was not finished, it is finished now.